test/e2e/helpers_test.go

```diff
@@ -66,15 +67,34 @@ const (
 )
 
 func getEnvironmentBuilder(ctx context.Context) (*environments.Builder, error) {
-	if existingCluster != "" {
+	if existingCluster == "" {
+		return createDefaultKINDBuilder(), nil
+	}
+
+	createCluster := func(ctx context.Context, cType string) (*environments.Builder, error) {
+		switch cType {
+		case string(gke.GKEClusterType):
+			return createDefaultGKEBuilder(ctx, clusterVersionStr)
+		case string(kind.KindClusterType):
+			return createDefaultKINDBuilder(), nil
+		default:
+			return nil, fmt.Errorf("unrecognized cluster type %s", cType)
+		}
+	}
+
+	clusterParts := strings.Split(existingCluster, ":")
```

Comment:
IMO it will be confusing to have `KONG_TEST_CLUSTER` aka `existingCluster` not pointing to an existing cluster. Can we have a separate env variable to tell the machinery to create a new cluster of a given type if `KONG_TEST_CLUSTER` is not set? e.g. `KONG_TEST_CLUSTER_PROVIDER`?

Reply:
Yes, that might make sense but in any case: the way this is written now, each test creates its own GKE cluster which is wasteful and takes forever. I need to bring back the cluster setup step somehow. 🤔